### ISSUE TITLE
Error prone warnings in com.google.template.soy.base.internal.Identifier

### ISSUE THREAD

**Issue by Yannic** [google/closure-templates#183]
When building `closure-templates` with enabled errorprone checks (2.3.1), I get the following error in `com.google.template.soy.base.internal.Identifier`:

```
bazel-out/<redacted>/java/src/com/google/template/soy/base/internal/_javac/internal/libinternal_sourcegenfiles/com/google/template/soy/base/internal/AutoValue_Identifier.java:8: error: [Immutable] Class extends @Immutable type com.google.template.soy.base.internal.Identifier, but is not immutable: 'AutoValue_Identifier' has non-final field 'type'
  private transient volatile Identifier.Type type;
                                             ^
    (see https://errorprone.info/bugpattern/Immutable)
  Did you mean 'private final transient volatile Identifier.Type type;'?
Target //<redacted>/java/src/com/google/template/soy/base/internal:internal failed to build
```

https://github.com/closure-tools/closure-templates/blob/7acea66c65ee0d7b4ceab2d18247cf8d8f15a437/java/src/com/google/template/soy/base/internal/BUILD.bazel#L23

/cc @robfig @sgammon

**Comment by iteriani** [comment]
Is this a bug in autovalue or is there a knob we can configure to generate this code?

**Comment by Yannic** [comment]
I'm not sure if it's a false positive or an actual bug in autovalue or soy. Someone more familiar with autovalue will have to decide that. What I do know is that you can add `@AutoValue.CopyAnnotations` and `@SuppressWarnings("Immutable")` to suppress just this one error: https://errorprone.info/bugpattern/Immutable#suppression

**Comment by Yannic** [comment]
I had a look at this again, and it seems like the cause for this error is that `type()` is lazily computed by annotating it as `@Memoized`. Per https://errorprone.info/bugpattern/Immutable, the fix is to add suppressions to the class.

Should I send a PR for this, or do you prefer fixing it yourself?

**Comment by sgammon** [comment]
@Yannic, @iteriani has been pretty quick to respond to PRs lately. happy to file one too :)

**Comment by Yannic** [comment]
@sgammon PRs need to be imported into Google, tested, submitted, and then pushed out. It's a simple 2 line change, so I'm not sure if it's worth the extra bit of effort if I submit the PR vs. submitting internally. Happy to do so if @iteriani wants me to.

**Comment by iteriani** [comment]
If you can submit it internally, just send me a CL. Otherwise, I end up
copy pasting PRs into google3 manually

**Comment by Yannic** [comment]
I'm not a Googler. I'll send a PR for copy&pasting.

**Comment by Yannic** [comment]
Fixed by 4f7808e7dfeed96b11d51d1534f4b13351b484d3
